docs/source/services/timelock_service/reverse-migration.rst

```diff
@@ -0,0 +1,140 @@
+
+Step 2: Take a Restore Timestamp
+--------------------------------
+
+Obtain a fresh timestamp for ``client`` from the TimeLock servers. This is important to ensure that the sequence
+of timestamps across TimeLock and the key-value service backed services is monotonically increasing.
+
+This can be obtained using ``curl`` as follows. Note that if you obtain an HTTP 503 error, this is because you curled a
+node that was not the leader; please try the other nodes.
+
+.. code-block:: none
+
+   $ curl localhost:8421/client/timestamp/fresh-timestamp
+   123456789
+
+Note down the value returned from this call; we'll refer to it as ``TS`` from here on.
+
+Step 3: Update your Key-Value Service
+-------------------------------------
+
+Update the timestamp table in your key value service, such that it is valid and has the bound set to ``TS`` .
+This will vary depending on your choice of key-value service.
+
+- If you are rolling back an automated migration with Cassandra or a migration using the steps outlined in
+  :ref:`Manual TimeLock Migration <manual-timelock-migration>`, then you can carry out the following steps.
+  We suppose that your clients have the keyspace ``client`` in the examples below; at a high level, we want to truncate
+  the timestamp table to remove all entries, and then add a single entry with row and column names ``ts`` and a blob
+  value indicating the actual timestamp.
+
+  .. code-block:: none
+
+     cqlsh> USE client;
+     cqlsh:client> SELECT * FROM "_timestamp" ;
+
+      key    | column1      | column2 | value
+     --------+--------------+---------+--------------------
+      0x7473 | 0x6f6c645473 |      -1 | 0x0000000006d30af4
+      0x7473 |       0x7473 |      -1 |               0x00
+
+     (2 rows)
+
+     cqlsh:client> TRUNCATE "_timestamp";
+     cqlsh:client> SELECT * FROM "_timestamp" ;
+
+      key | column1 | column2 | value
+     -----+---------+---------+-------
+
+     (0 rows)
+
+     cqlsh:client> INSERT INTO "_timestamp" (key, column1, column2, value) VALUES (0x7473, 0x7473, -1, 0x00000000075bcd15);
+     cqlsh:client> SELECT * FROM "_timestamp" ;
+
+      key    | column1 | column2 | value
+     --------+---------+---------+--------------------
+      0x7473 |  0x7473 |      -1 | 0x00000000075bcd15
+
+  You will need to encode ``TS`` into an 8-byte hex representation; this can be done using the ``printf`` command:
+
+  .. code-block:: none
+
+     $ printf '0x%016x\n' 123456789
+     0x00000000075bcd15
+
+- If you are using DBKVS and have followed the steps outlined in :ref:`Manual TimeLock Migration<manual-timelock-migration>`,
+  it suffices to rename the column back:
+
+  .. code:: sql
+
+     ALTER TABLE atlasdb_timestamp RENAME LEGACY_last_allocated TO last_allocated;
+
+Step 4: Revert AtlasDB Client Configurations
+--------------------------------------------
+
+Change the configuration on your AtlasDB clients to not reference TimeLock any longer. For more detail on options
```

Comment:
Sure, changed to "Remove the timelock block from your AtlasDB client configurations."